src/AutoBalance.cpp

```diff
@@ -407,7 +407,19 @@ class AutoBalance_AllMapScript : public AllMapScript
 
             AutoBalanceMapInfo *mapABInfo=map->CustomData.GetDefault<AutoBalanceMapInfo>("AutoBalanceMapInfo");
 
-            mapABInfo->playerCount--;// (maybe we've to found a safe solution to avoid player recount each time)
+            if ((map->GetEntry()->IsDungeon() || map->GetEntry()->IsRaid()) && player) {
+              Map::PlayerList const& playerList = map->GetPlayers();
+              for (Map::PlayerList::const_iterator playerIteration = playerList.begin(); playerIteration != playerList.end(); ++playerIteration) {
+                if (playerIteration->GetSource()->IsAlive() && !playerIteration->GetSource()->IsInCombat()) { 
+                  mapABInfo->playerCount--;
+                } else if (Player* playerHandle = playerIteration->GetSource()) {
+                  ChatHandler chatHandle = ChatHandler(playerHandle->GetSession());
+                  chatHandle.PSendSysMessage("|cffFF0000 [-AutoBalance]|r|cffFF8000 %s has left %s During Combat, Re-Enter to Fix Scaling.|r", player->GetName().c_str(), map->GetMapName());
```

Comment:
reenter to fix scaling???

if someone leaves during combat, he is certainly not gonna come back to fix the other group's scaling

Reply:
how else can we fix this, without players rejoin instance?

Reply:
i don't know, maybe @Helias can help, maybe test ingame before of how inconvenient it is or not